src/mlpack/core/metrics/lmetric.hpp

```diff
@@ -116,6 +116,10 @@ typedef LMetric<2, true> EuclideanDistance;
  */
 typedef LMetric<INT_MAX, false> ChebyshevDistance;
 
+/**
+ * The HDBSCAN metric.
+ */
+typedef LMetric<INT_MAX, true> HdbscanDistance;
```

Comment:
Instead of doing this, if I define the metric in namespace HDBSCAN{} , the user will not be able to select the treeType because of "static_assert(meta::IsLMetric<MetricType>::Value == true" in hrectbound.hpp .